/**
 * @file kliker.h
 * @brief Игра "Кликер" - заработок робуксов кликами
 * @author Development Team
 * @date 2025
 * @version 1.0
 * 
 * Этот файл содержит объявление класса Kliker, который реализует
 * простую кликер-игру для заработка внутриигровой валюты (робуксов).
 * Игра позволяет пользователям зарабатывать робуксы без риска,
 * кликая по специальной кнопке.
 */

#ifndef KLIKER_H
#define KLIKER_H

#include <QDialog>
#include "clientapi.h"

namespace Ui {
class Kliker;
}

/**
 * @brief Игра "Кликер" - безрисковый заработок робуксов
 * 
 * Класс Kliker реализует простую кликер-игру, где пользователи могут
 * зарабатывать робуксы без риска потери средств. Игровая механика:
 * - Клик по кнопке приносит фиксированное количество робуксов
 * - Отсутствие ставок и риска потери средств
 * - Простой и интуитивный интерфейс
 * - Мгновенное начисление заработанных робуксов на баланс
 * 
 * Это альтернативный способ пополнения баланса для игроков,
 * которые предпочитают избегать азартных игр.
 */
class Kliker : public QDialog
{
    Q_OBJECT

public:
    /**
     * @brief Конструктор игры "Кликер"
     * @param parent Родительский виджет
     */
    explicit Kliker(QWidget *parent = nullptr);
    
    /**
     * @brief Деструктор игры "Кликер"
     */
    ~Kliker();

    /**
     * @brief Показать окно игры
     * 
     * Отображает интерфейс кликера и подготавливает элементы
     * управления для взаимодействия с пользователем
     */
    void slot_show();

private slots:
    /**
     * @brief Обработчик кнопки "Назад"
     * 
     * Возвращает игрока в главное меню казино
     */
    void on_pushButton_clicked();

    /**
     * @brief Обработчик кнопки заработка робуксов
     * 
     * Основная игровая механика:
     * - Начисляет фиксированное количество робуксов за клик
     * - Отправляет запрос на сервер для обновления баланса
     * - Обновляет отображение заработанных средств
     * - Применяет ограничения по частоте кликов (если есть)
     */
    void on_pushButtonTap_clicked();

private:
    Ui::Kliker *ui;          ///< Пользовательский интерфейс игры

    int Robux100;            ///< Счетчик заработанных робуксов за сессию

    ClientApi *clientApi;    ///< API для взаимодействия с сервером

signals:
    /**
     * @brief Сигнал возврата в главное меню
     * 
     * Испускается при завершении игровой сессии и возврате
     * в основное меню казино
     */
    void to_main();
};

#endif // KLIKER_H
